### PR TITLE
Code verification problems for 1D heat conduction in each coordinate system

### PR DESCRIPTION
**Pull request by atoptann**
Ref #15301

### CODE REVIEW

**Comment by atoptann** [comment]
@bwspenc

**Comment by atoptann** [comment]
@gambka

**Comment by loganharbour** [comment]
@atoptann you've been added to the idaholab user group so your testing should start automatically now.

**Comment by atoptann** [comment]
thanks @loganharbour

**Review by andrsd — DISMISSED** [review]

File: modules/heat_conduction/test/tests/code_verification/tests [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Tests]
+  issues = '#15301'
+  design = 'HeatConduction.md'
+  [spatial_csv]
+    type = CSVDiff
+    command = mes_spatial.py
+    recover = false # this is not running a MooseApp, recover doesn't make any sense
+    csvdiff = 'cartesian_test_no1_first_order.csv'
+    requirement = "The system shall include a tool for writing spatial convergence data to a file using method of exact solutions."
+    required_python_packages = 'sympy'
```
[the remark this answers is not in the document]
Comment (atoptann) [reply modules/heat_conduction/test/tests/code_verification/tests:11]: I added now. thank you for your review.

**Comment by andrsd** [comment]
@atoptann I requested review from @gambka and @bwspenc since you tagged them. I don't know if you can actually do that via github interface.

I am fine with the fixes you did that I requested.

File: modules/heat_conduction/test/tests/code_verification/cartesian_test_no2.i [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# Problem I.2
+#
+# The thermal conductivity of an infinite plate defined varies linearly
```
Comment (gambka) [inline modules/heat_conduction/test/tests/code_verification/cartesian_test_no2.i:3]: An infinite plate with a thermal conductivity that varies linearly with temperature?

File: modules/heat_conduction/test/tests/code_verification/cylindrical_test_no1.i [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# Problem II.1
+#
+# An infinitely long hollow cylinder has an inner radius ri and
+# outer radius ro. It has a constant thermal conductivity k and
+# internal heat generation q. It is allowed to reach thermal
+# equilibrium while being exposed constant temperatures on its
```
Comment (gambka) [inline modules/heat_conduction/test/tests/code_verification/cylindrical_test_no1.i:6]: while being exposed **to** constant temperatures

**Review by gambka — CHANGES_REQUESTED** [review]
Is there a plan to include .md documentation similar to your report that details the problems and includes plots of the results?

**Comment by atoptann** [comment]
> Is there a plan to include .md documentation similar to your report that details the problems and includes plots of the results?

Not at the moment. After the journal article is published.

**Comment by andrsd** [comment]
I have one more thing here. The test takes 24secs on my machine. That's way too long for a regression test. Our rule of thumb is it needs to finish under 3secs. So you have 2 options:

1. have this as a heavy test:
    you would say:
    ```
    heavy = true
    ```
    in the `tests` file

2. you would break each run into a single test that finishes under 3 secs and checks the convergence rate. You would have to do something about generating the final graph after all the tests are done, if that's really needed.

However, I don't know what is the intent with these tests, i.e. either regression or something else like what BISON does for assessment or something else.

**Comment by atoptann** [comment]
> I have one more thing here. The test takes 24secs on my machine. That's way too long for a regression test. Our rule of thumb is it needs to finish under 3secs. So you have 2 options:
> 
> 1. have this as a heavy test:
>    you would say:
>    ```
>    heavy = true
>    ```
>    
>    
>    in the `tests` file
> 2. you would break each run into a single test that finishes under 3 secs and checks the convergence rate. You would have to do something about generating the final graph after all the tests are done, if that's really needed.
> 
> However, I don't know what is the intent with these tests, i.e. either regression or something else like what BISON does for assessment or something else.

The first option is fine. I prefer them to be in the same place. I pushed the changes now.

**Comment by bwspenc** [comment]
How fast do the tests run now? 
I thought the rule was that they have to run in <1 second, but maybe that depends on who you ask.

**Comment by andrsd** [comment]
> How fast do the tests run now?

Which ones? These new ones? The ones in `heat_conduction`? Or those in `modules`?

> I thought the rule was that they have to run in <1 second, but maybe that depends on who you ask. 

For regression tests you want as short as possible. But, to me those are not regression tests really. If you guys are testing grid convergence rate, then to break that you would need to break some fundamental stuff in MOOSE, not in the `heat_conduction` module.

**Comment by atoptann** [comment]
Test failures are irrelevant and unfortunately hold this PR to be merged. No code changes are made, only new verification tests are added.

**Comment by loganharbour** [comment]
Should be good in ~an hour. will invalidate then.

**Comment by atoptann** [comment]
thanks @loganharbour

**Review by bwspenc — APPROVED** [review]

**Pull request merged**